adapter/pkg/discovery/cache/v3/simple.go

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	envoy_cache "github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/log"
+)
+
+// SnapshotCache is a snapshot-based cache that maintains a single versioned
+// snapshot of responses per node. SnapshotCache consistently replies with the
+// latest snapshot. For the protocol to work correctly in ADS mode, EDS/RDS
+// requests are responded only when all resources in the snapshot xDS response
+// are named as part of the request. It is expected that the CDS response names
+// all EDS clusters, and the LDS response names all RDS routes in a snapshot,
+// to ensure that Envoy makes the request for all EDS clusters or RDS routes
+// eventually.
+//
+// SnapshotCache can operate as a REST or regular xDS backend. The snapshot
+// can be partial, e.g. only include RDS or EDS resources.
+type SnapshotCache interface {
```

Comment:
Remove interface def and use envoy interface.

Reply:
Can remove this since we are using our Own `Snapshot` struct. Thats different from the struct expected from the envoy `SnapshotCache`